scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/constraints/filters/FilterEverViolated.java

```diff
@@ -1,23 +1,24 @@
 package gov.nasa.jpl.aerie.scheduler.constraints.filters;
 
+import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
 import gov.nasa.jpl.aerie.constraints.time.Window;
 import gov.nasa.jpl.aerie.constraints.time.Windows;
+import gov.nasa.jpl.aerie.constraints.tree.Expression;
 import gov.nasa.jpl.aerie.scheduler.model.Plan;
-import gov.nasa.jpl.aerie.scheduler.constraints.resources.StateConstraintExpression;
 
 /**
  * filter in intervals if constraint expression @expr is ever violated during it
  */
 public class FilterEverViolated extends FilterFunctional {
 
-  private final StateConstraintExpression expr;
+  private final Expression<Windows> expr;
 
-  public FilterEverViolated(StateConstraintExpression expr) {
+  public FilterEverViolated(final Expression<Windows> expr) {
     this.expr = expr;
   }
 
   @Override
-  public boolean shouldKeep(Plan plan, Window range) {
-    return !(expr.findWindows(plan, new Windows(range)).equals(new Windows(range)));
+  public boolean shouldKeep(final SimulationResults simulationResults, final Plan plan, final Window range) {
+    return !(expr.evaluate(simulationResults, range, null).equals(new Windows(range)));
```

Comment:
The "empty" case for environment is an empty map, not `null`.
```suggestion
    return !(expr.evaluate(simulationResults, range, Map.of()).equals(new Windows(range)));
```